data/mods/gen2/rulesets.ts

```diff
@@ -33,4 +33,99 @@ export const Rulesets: {[k: string]: ModdedFormatData} = {
 			'Spore + Spider Web',
 		],
 	},
+	nintendocup2000movelegality: {
+		effectType: 'ValidatorRule',
+		name: 'Nintendo Cup 2000 Move Legality',
+		desc: "Prevents Pok\u00e9mon from having moves that would only be obtainable in Pok\u00e9mon Crystal.",
+		onValidateSet(set) {
+			const illegalCombos: {[speciesid: string]: {[moveid: string]: 'E' | 'L' | 'S'}} = {
+				arbok: {crunch: 'E'},
+				sandslash: {metalclaw: 'E'},
+				golduck: {crosschop: 'E'},
+				marowak: {swordsdance: 'E'},
+				electabuzz: {crosschop: 'E'},
+				magmar: {crosschop: 'E'},
+				jolteon: {batonpass: 'L'},
+				vaporeon: {batonpass: 'L'},
+				flareon: {batonpass: 'L'},
+				espeon: {batonpass: 'L'},
+				umbreon: {batonpass: 'L'},
+				dragonite: {extremespeed: 'S'},
+				meganium: {swordsdance: 'E'},
+				typhlosion: {submission: 'E'},
+				ariados: {agility: 'L'},
+				yanma: {wingattack: 'L'},
+				murkrow: {skyattack: 'E'},
+				qwilfish: {spikes: 'L'},
+				sneasel: {metalclaw: 'L'},
+				ursaring: {metalclaw: 'E'},
+				piloswine: {amnesia: 'L'},
+				skarmory: {skyattack: 'E'},
+				donphan: {watergun: 'E'},
+				suicune: {aurorabeam: 'L'},
+				dugtrio: {triattack: 'L'},
+				magneton: {triattack: 'L'},
+				cloyster: {spikes: 'L'},
+			};
+			const notUsableAsTM = ['icebeam', 'flamethrower', 'thunderbolt'];
+			const species = this.dex.species.get(set.species || set.name);
+			const learnsetData = {...(this.dex.data.Learnsets[species.id]?.learnset || {})};
+			const legalityList = illegalCombos[species.id];
+			if (!legalityList) return;
+			const problems = [];
+			let prevo = species.prevo;
+			while (prevo) {
+				const prevoSpecies = this.dex.species.get(prevo);
+				const prevoLsetData = this.dex.data.Learnsets[prevoSpecies.id]?.learnset || {};
+				for (const moveid in prevoLsetData) {
+					if (!(moveid in learnsetData)) {
+						learnsetData[moveid] = prevoLsetData[moveid];
+					} else {
+						learnsetData[moveid].push(...prevoLsetData[moveid]);
```

Comment:
Oh, hey, this crash is from the same PR! I somehow missed this in my big refactor because it was reading `data.Learnsets` directly instead of using one of the accessor methods.